### ISSUE TITLE
Service is active (running) after main process has exited

### ISSUE THREAD

**Issue by ElvishJerricco** [systemd/systemd#28030]
### systemd version the issue has been seen with

253.5

### Used distribution

NixOS 23.05

### Linux kernel version used

6.1.33

### CPU architectures issue was seen on

x86_64

### Component

systemd

### Expected behaviour you didn't see

A service unit with `Type=simple` and `ExitType=main` should not be considered `active (running)` after the main process has exited.

### Unexpected behaviour you saw

The service unit is still `active (running)` after the main process has exited.

### Steps to reproduce the problem

Start this service:

```
[Service]
Type=simple
ExitType=main
ExecStart=/usr/bin/bash -c "/usr/bin/systemctl --version; /usr/bin/sleep infinity &"
```

It should immediately become `inactive (dead)`, as the main bash process exits immediately. Instead, it's `active (running)`.

I've bisected this issue down to this commit: https://github.com/systemd/systemd/commit/ae83e97a51519ca33e70d7ba142cb3ed24212825, which is the backport of this PR: https://github.com/systemd/systemd/pull/27735

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
● show-bug.service
     Loaded: loaded (/etc/systemd/system/show-bug.service; enabled; preset: enabled)
     Active: active (running) since Wed 2023-06-14 00:58:36 UTC; 3s ago
    Process: 702 ExecStart=/usr/bin/bash -c /usr/bin/systemctl --version; /usr/bin/sleep infinity & (code=exited, status=0/SUCCESS)
   Main PID: 702 (code=exited, status=0/SUCCESS)
         IP: 0B in, 0B out
         IO: 0B read, 0B written
      Tasks: 1 (limit: 1150)
     Memory: 404.0K
        CPU: 4ms
     CGroup: /system.slice/show-bug.service
             └─717 /usr/bin/sleep infinity

Jun 14 00:58:36 machine systemd[1]: Started show-bug.service.
Jun 14 00:58:36 machine bash[703]: systemd 253 (253.5)
Jun 14 00:58:36 machine bash[703]: +PAM +AUDIT -SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT -GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT -QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK -XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified
```

**Comment by Toolybird** [comment]
Essentially the same issue as #27953

**Comment by msekletar** [comment]
Hi @ElvishJerricco I've posted fix for #27953 (which I think is duplicate of this one) as PR #28035. Testing that PR would be much appreciated.

**Comment by ElvishJerricco** [comment]
Closed by #28000
